docs/use_cases/multi_agent_rag.md

```diff
@@ -0,0 +1,279 @@
+
+The ranker agent evaluates the relevance of the retrieved passages using additional ranking signals like source credibility, passage specificity, and lexical overlap. This provides a relevance-based filtering step. This agent might be using ontology, for example, as a way to rerank retrieved information. 
+
+The reader agent summarizes lengthy retrieved passages to succinct snippets containing only the most salient information. This distills the context down to key facts.
+
+Finally, the orchestrator agent dynamically adjusts the relevance weighting and integration of the prompt and filtered, ranked context passages (i.e., prompt hybridization) to maximize coherence in the final augmented prompt.
+
+## Benefits of multi-agent RAG architecture
+
+- Agent-specific, focused specialization _improves relevance and quality_. Retriever agents leverage tailored similarity metrics, rankers weigh signals like source credibility, and readers summarize context.
+- Asynchronous operation _reduces latency_ by parallelizing retrieval. Slow operations don't block faster ones.
+- Salient extraction and abstraction techniques achieve _better summarization_. Reader agents condense complex information from retrieved passages into concise, coherent, highly informative summaries.
+- Prompt hybridization achieves _optimized prompting_. Orchestrator agents balance prompt and ranked context data for more coherent outcome prompts.
+- Flexible, modular architecture _enables easy horizontal scaling_ and optional _incorporation of new data sources_. You can enhance iteratively over time by adding more agents (e.g., a visualizer agent to inspect system behavior), or substituting alternative implementations of any agent.
+
+These benefits of multi-agent over single-agent RAG have been demonstrated in experiments [any examples?].
+
+
+Let's look at an implementation of multi-agent RAG, and then explore in more depth the range of possible optimizations made possible by multi-agent RAG.
+
+## Example with Autogen library: https://github.com/microsoft/autogen
+
+1. AssistantAgent: They are given a name, a system message, and a configuration object (llm_config). The system message is a string that describes the role of the agent. The llm_config object is a dictionary that contains functions for the agent to perform its role.
+
+2. user_proxy is an instance of UserProxyAgent. It is given a name and several configuration options. The is_termination_msg option is a function that determines when the user wants to terminate the conversation. The human_input_mode option is set to "NEVER", which means the agent will never ask for input from a human. The max_consecutive_auto_reply option is set to 10, which means the agent will automatically reply to up to 10 consecutive messages without input from a human. The code_execution_config option is a dictionary that contains configuration options for executing code.
```

Comment:
2. Nothing described appears in the code snippet below - e.g., 
user_proxy, UserProxyAgent, is_termination_msg, human_input_mode, max_consecutive_auto_reply, code_execution_config